### PR TITLE
Fix parsing map keys in `fromBalString` method

### LINKED ISSUE #41700: [Bug]: The `fromBalString` function is not working properly for map expression
### Description

$ Subject

### Steps to Reproduce

Run the below code:
```ballerina
import ballerina/io;

public function main() returns error? {
    map<string> _ = {name:"John", city:"London"};

    string mapExpString = string `{name: "John", city: "London"}`;
    anydata output = check mapExpString.fromBalString();
    // Expected map: {"name":"John", "city":"London"}
    io:println(output);
}
```

```console
$ bal run sample.bal

Compiling source
        tharmigan/sample1:0.1.0

Running executable

{"am":"John","cit":"London"}
```

### Affected Version(s)

Ballerina SwanLake 2201.7.x, 2201.8.x

### OS, DB, other environment details and versions

_No response_

### Related area

-> Runtime

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by Nadeeshan96**
$title
Fixes #41700

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: langlib/langlib-test/src/test/resources/test-src/valuelib_fromBalString_test.bal [hunk 1]
```diff
@@ -163,8 +163,97 @@ function testMapFromBalString() {
     }
 
     assert(result3, mapVal3);
-}
 
+    string mapExpString = string `{"name":"John","city":"London"}`;
```
Comment (Nadeeshan96) [inline langlib/langlib-test/src/test/resources/test-src/valuelib_fromBalString_test.bal:167]: Add a test with arrays as values

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/BalStringUtils.java [hunk 2]
```diff
@@ -185,6 +187,17 @@ public static Object parseMapExpressionStringValue(String exprValue, BLink paren
         }
     }
 
+    private static String getMapKey(String e, int colonIndex) {
+        String key = e.substring(0, colonIndex).trim();
+        if (key.startsWith("\"") && key.endsWith("\"")) {
+            key = key.substring(1, key.length() - 1);
+        } else {
+            throw ErrorCreator.createError(fromString("invalid expression style string value. " +
+                    "map keys are not enclosed with '\"'."));
```
[the remark this answers is not in the document]
Comment (Nadeeshan96) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/BalStringUtils.java]: Changed the error message with `:`.

File: langlib/langlib-test/src/test/resources/test-src/valuelib_fromBalString_test.bal [hunk 3]
```diff
@@ -163,8 +163,107 @@ function testMapFromBalString() {
...
+
+    mapExpString = string `
+              {                 name"  :
+                3 ,
+                         city" : 12.55      }`;
+    output = mapExpString.fromBalString();
+    assert(output is error, true);
+    if (output is error) {
+        assert("invalid expression style string value. map keys are not enclosed with '\"'.",
+        <string>checkpanic output.detail()["message"]);
```
[the remark this answers is not in the document]
Comment (Nadeeshan96) [reply langlib/langlib-test/src/test/resources/test-src/valuelib_fromBalString_test.bal:264]: Have added a test.

**Review by warunalakshitha — APPROVED** [review]
LGTM

**Pull request merged**